src/dotnet/ShellShim/ShellShimPath.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ShellShim
+{
+    internal class ShellShimPath
```

Comment:
extract class to move all path stuff here. And it is easy to use global vs bin-path

Reply:
:+1:

Reply:
other member in ShellShimRepository don't need GetPosixShimPath. Also by extra class also make ShellShimRepository's responsibility more stand out.